### ISSUE TITLE
Go external lang server support is broken for bingo

### ISSUE THREAD

**Issue by illegalnumbers** [Microsoft/vscode-go#2394]
See https://github.com/saibing/bingo/issues/159 as I think it is related. The version of bingo I have locally is from March 3 2019 so I don't believe the issue is in bingo per se but with some other change in vscode that has led to the external language server not working. My config is listed below. If I enable an external server the go to definition option is totally removed from vscode.

```
 "go.gopath": "<mygopath> 
 "go.docsTool": "gogetdoc",
    "go.formatTool": "gofmt",
        "go.useLanguageServer": true,
        "go.alternateTools": {
            "go-langserver": "bingo"
        },
        "go.languageServerFlags": [],
       "go.languageServerExperimentalFeatures": {
        "format": true,
        "autoComplete": true,
        "rename": true,
        "goToDefinition": true,
        "hover": true,
        "signatureHelp": true,
        "goToTypeDefinition": true,
        "goToImplementation": true,
        "documentSymbols": true,
        "workspaceSymbols": true,
        "findReferences": true
    },
```
Steps to Reproduce:

1. configure as above
2. right click or ctrl+click to go to definition
3. notice nothing is there

**Comment by dsfish** [comment]
@illegalnumbers, I don't know if this is a new issue, but I typically need to wait ~90s for the language server (bingo) to load before those options appear. You should get a dialog box in the bottom-right corner saying that the language server has loaded and then they will be available. Are the options still missing several minutes after opening VS Code?

**Comment by illegalnumbers** [comment]
Yea it is. I'm gonna try updating to go 1.12 and updating bingo to be more recent and see if it still persists.

**Comment by illegalnumbers** [comment]
Yea same result `unexpected panic: bad uri "file://" (path "" MUST have leading slash; it can't be relative)`

**Comment by illegalnumbers** [comment]
It also prompts to update go packages everytime vscode opens

**Comment by illegalnumbers** [comment]
It actually seems like bingo isnt even running as I think it is this exception https://github.com/sourcegraph/go-langserver/blob/master/langserver/fs.go#L211

**Comment by illegalnumbers** [comment]
Removing `go.useLanguageServer` seems to have allowed the go to definition right click option to appear but I still cant actually do that.

**Comment by dsfish** [comment]
That makes sense. Good luck, I'm debugging my own (separate) bingo issues right now 😞

**Comment by illegalnumbers** [comment]
Hmm ok well the panic when I look in the application logs actually starts here `/go/src/github.com/saibing/bingo/langserver/internal/util/util.go:116` seemingly

**Comment by illegalnumbers** [comment]
https://github.com/saibing/bingo/blob/master/langserver/handler_shared.go#L23 --> this error

**Comment by OneOfOne** [comment]
If you're using go from tip, try `export GO111MODULE=off` before starting `vscode` from the terminal.

**Comment by illegalnumbers** [comment]
I'm also using go modules so that doesn't really help. Turning that on at least gets it so it doesnt panic immediately though.

**Comment by ramya-rao-a** [comment]
@illegalnumbers 
- Select `View` from the menu, then `Output`, then `go-langserver` from the drop down on the top right corner of the output panel. This is where logs from the language server will appear. Check if there are any errors there
- Set `go.languageServerFlags` to `["-trace"]` and then reload VS Code. Look for errors in the logs as instructed in the previous steps. 

The above should help you get the logs from the language server and help in figuring out what is happening with bingo.

Regardless, I don't believe this is an issue with VS Code or the Go extension.
To confirm, you can try [older versions of VS Code](https://code.visualstudio.com/updates/v1_32) or the [older versions of the Go extension](https://github.com/Microsoft/vscode-go/releases)

Please create an issue in the repo for bingo. Maybe @saibing can help in figuring out what is happening.

You can also give the [Go language server from Google](https://github.com/golang/go/wiki/gopls) a try
